src/client/create-http-client.ts

```diff
@@ -45,7 +46,18 @@ export function createHttpClient<S extends HttpSchema>(
       .compile(path)(info?.params)
       .replace(/\*/g, () => info?.params[i++]);
 
-    return axiosClient({ method, url, data: info?.body });
+    // here we pick the schema definition from given method & path
+    // then we'll parse the response payload
+    // this will ensure we have correct types in the Frontend (e.g.: Date instance, not a string of date)
+    const responseBodySchema = schema[`${method} ${path}`].responseBody;
+
+    return axiosClient({ method, url, data: info?.body }).then((response) => {
+      // if we fail to parse here, mean our API is returning something weird
+      // an Exception would be thrown by Zod
+      response.data = responseBodySchema.parse(response.data);
```

Comment:
Does this introduce any performance issues?

Reply:
Likely, but since we all are fans of **strictly typed**, there would be consequences. 

Otherwise, we won't have the correct types on the client side 😢